progressbar2 4.2.0

skupr-anaconda opened the pull request:
**Jira ticket:** []() (-)

**The upstream data:**
Changelog: https://github.com/WoLpH/python-progressbar/releases
License: https://github.com/wolph/python-progressbar/blob/v4.2.0/LICENSE
Requirements: https://github.com/wolph/python-progressbar/blob/v4.2.0/setup.py

**_Actions:_**

- Skip `py<37`

- Fix `script`

- Upate host and run dependencies

- Add pip check

- Update description

**_Notes:_**

 * `grayskull` requires `progressbar2 >=3.53.0`

**Additional info:**

<details>

**Package's statistics**

<details>

 * Priority D | effort: easy | Category: other | subcategory:  | pkg_type: python
 * Outdated platfroms: 'linux-64', 'linux-aarch64', 'linux-ppc64le', 'osx-64', 'osx-arm64', 'win-64'
 * Latest version: 4.2.0
 * Release on PyPi:
    * version: 4.2.0
    * date: 2022-10-26T13:47:49
* [PyPi history](https://pypi.org/project/progressbar2/#history)
 * Popularity: 
    * 3 months downloads: 16405.0
    * All time:  722629 downloads -  progressbar2  4.2.0  A Python Progressbar library to provide visual (yet text based) progress to long running operations.  

</details>

**Other checks:**

<details>

1. - [x] Check the pinnings
2. - [x] Verify that the `build_number` is correct
3. - [x] has `setuptools`
6. - [x] Verify if the package needs `wheel`
    
8. - [x] `pip` in test
    
9. - [x] Verify the test section
10. - [x] Verify if the package is `architecture specific`
11. - [x] Verify that private modules are not mentioned in the recipe For example: (_private_module)
12.  - [x] license_file: LICENSE is present

14. - [x] license_family BSD is present
16. - [x] License: BSD-3-Clause
    - [x] License is `spdx` compliant
 * Check if the license identifier has a correct name from the SPDX License List

| Identifier                           |
|:-------------------------------------|
| BSD-3-Clause                         |
| BSD-3-Clause-Attribution             |
| BSD-3-Clause-Clear                   |
| BSD-3-Clause-LBNL                    |
| BSD-3-Clause-Modification            |
| BSD-3-Clause-No-Military-License     |
| BSD-3-Clause-No-Nuclear-License      |
| BSD-3-Clause-No-Nuclear-License-2014 |
| BSD-3-Clause-No-Nuclear-Warranty     |
| BSD-3-Clause-Open-MPI                |
</details>

**Check dependency issues:**

<details>
../aggregate/progressbar2-feedstock/recipe/meta.yaml
Dependencies: ['pytest-runner >=2.8', 'python-utils', 'python', 'setuptools', 'six']


noarch:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

linux-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

linux-ppc64le:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

linux-s390x:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

osx-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

osx-arm64:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

win-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

</details>

**Package Build Score: 18**


**Links:**
* [Anaconda Recipes feedstock](https://github.com/AnacondaRecipes/progressbar2-feedstock)
* [Update branch](https://github.com/AnacondaRecipes/progressbar2-feedstock/tree/4.2.0)
* [conda-forge recipe](https://github.com/conda-forge/{feedstock_name}-feedstock)
* [PyPI](https://pypi.org/project/progressbar2)
* [Diff between upstream and feature branch](https://github.com/conda-forge/progressbar2-feedstock/compare/main...AnacondaRecipes:4.2.0)
* [Diff between origin and feature branch](https://github.com/AnacondaRecipes/progressbar2-feedstock/compare/master...AnacondaRecipes:4.2.0)
* [The last merged PRs](https://github.com/AnacondaRecipes/progressbar2-feedstock/pulls?q=is%3Apr+is%3Amerged+sort%3Aupdated-desc+)


</details>

**Updating the recipe:**
If the recipe needs additional modification the update branch can be modified. Note that the PR diffs are not updated with these changes.
```
git clone -b 4.2.0 git@github.com:AnacondaRecipes/progressbar2-feedstock.git
```

skupr-anaconda committed:
DOR update to 4.2.0

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_c43_enui5z/clone/recipe/meta.yaml:16: uses_setup_py: `python setup.py install` is deprecated.
ERROR: /tmp/abs_c43_enui5z/clone/recipe/meta.yaml:22: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_c43_enui5z/clone/recipe/meta.yaml:23: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_c43_enui5z/clone/recipe/meta.yaml:23: missing_wheel: For pypi packages, wheel should be present in the host section
ERROR: /tmp/abs_c43_enui5z/clone/recipe/meta.yaml:32: missing_pip_check: For pypi packages, pip check should be present in the test commands
INFO: /tmp/abs_c43_enui5z/clone/recipe/meta.yaml:67: invalid_url: http://progressbar-2.readthedocs.org/en/latest/ : URL domain redirect progressbar-2.readthedocs.org ->  progressbar-2.readthedocs.io
ERROR: /tmp/abs_c43_enui5z/clone/recipe/meta.yaml:68: http_url: http://progressbar-2.readthedocs.org/en/latest/ is not https
Errors were found

</details>

skupr-anaconda committed:
Release 4.2.0

- Skip py<37

- Fix script

- Upate host and run dependencies

- Add pip check

- Update description

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: /tmp/abs_6eorargtvq/clone/recipe/meta.yaml:71: invalid_url: http://progressbar-2.readthedocs.org/en/latest/ : URL domain redirect progressbar-2.readthedocs.org ->  progressbar-2.readthedocs.io
ERROR: /tmp/abs_6eorargtvq/clone/recipe/meta.yaml:72: http_url: http://progressbar-2.readthedocs.org/en/latest/ is not https
Errors were found

</details>

katietz reviewed (approved):
lgtm. Especially the catch about replacing python by  {{ PYTHON }}

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: /tmp/abs_ceqx5c0_tz/clone/recipe/meta.yaml:71: invalid_url: http://progressbar-2.readthedocs.org/en/latest/ : URL domain redirect progressbar-2.readthedocs.org ->  progressbar-2.readthedocs.io
ERROR: /tmp/abs_ceqx5c0_tz/clone/recipe/meta.yaml:72: http_url: http://progressbar-2.readthedocs.org/en/latest/ is not https
Errors were found

</details>
